src/source/PeerConnection/SessionDescription.c

```diff
@@ -316,6 +326,64 @@ PCHAR fmtpForPayloadType(UINT64 payloadType, PSessionDescription pSessionDescrip
     return NULL;
 }
 
+/*
+ * Extracts a (hex) value after the provided prefix string. Returns true if
+ * successful.
+ */
+static BOOL readHexValue(const PCHAR input, const PCHAR prefix, PINT32 value) {
+  const PCHAR substr = STRSTR(input, prefix);
+  if (substr != NULL) {
+    if (sscanf(substr + STRLEN(prefix), "%x", value) == 1) {
+      return TRUE;
+    }
+  }
+  return FALSE;
+}
+
+/*
+ * Scores the provided fmtp string based on this library's ability to
+ * process various types of H264 streams. A score of 0 indicates an
+ * incompatible fmtp line. Beyond this, a higher score indicates more
+ * compatibility with the desired characteristics, packetization-mode=1,
+ * level-asymmetry-allowed=1, and inbound match with our preferred
+ * profile-level-id.
+ */
+INT32 getH264FmtpScore(PCHAR fmtp) {
+  INT32 profileId = 0, packetizationMode = 0, levelAsymmetry = 0;
+  BOOL isAsymmetryAllowed = FALSE;
+  BOOL isProfileMatch = FALSE;
+  INT32 score = 0;
+
+  // No ftmp match found.
+  if (fmtp == NULL) {
+      return 0;
+  }
+
+  // Currently, the packetization mode must be 1, as the packetization logic
+  // is currently not configurable, and sends both NALU and FU-A packets.
+  // https://tools.ietf.org/html/rfc7742#section-6.2
+  if (readHexValue(fmtp, "packetization-mode=", &packetizationMode) == FALSE || packetizationMode != 1) {
+      return 0;
+  }
+
+  if (readHexValue(fmtp, "profile-level-id=", &profileId) == TRUE) {
+      isProfileMatch = profileId == 0x42e01f;
+  }
+
+  if (readHexValue(fmtp, "level-asymmetry-allowed=", &levelAsymmetry) == TRUE) {
+      isAsymmetryAllowed = (levelAsymmetry == 1);
+  }
+
+  if (isProfileMatch) {
```

Comment:
Seems that you could use uint32s instead of these booleans and simply return the addition

Reply:
How strongly do you feel about this? I just tried it out with that approach, and it felt a bit less intuitive/readable to someone trying to intuit what the scoring meant...

Reply:
Good documentation is the key. This approach is better future proof in terms of getting even closer match